### ISSUE TITLE
MQTT.js dependency is outdated

### ISSUE THREAD

**Issue by celalo** [aws/aws-iot-device-sdk-js#103]
Currently the project is depending on MQTT.js v1.11.2 which was released 6 months ago. Current MQTT.js version is 2.1.3 and v2.0.0 is claimed to be 3x faster in sending packets.

When can we expect this dependency to be up to date?

**Comment by chaurah** [comment]
Hi @celalo,
     We are already working on updating to the newer versions of our dependencies in the next release of the SDK. I don't have a time-frame for release at the moment unfortunately. We will update this issue as soon as the next release is out which contains updated versions.

Please do let us know if you have any further suggestions. Thank you for using AWS IoT.

Rahul

**Comment by gkwicker** [comment]
Hi @celalo, you can use [npm shrinkwrap](https://docs.npmjs.com/cli/shrinkwrap) to force aws-iot-device-sdk to use version 2.1.3 of MQTT.js in your package.  By placing an `npm-shrinkwrap.json` file in the directory where your `package.json` is located, npm will use the dependency versions you've specified in there.  Here's an example `npm-shrinkwrap.json` file for a package named `test-package` which will force aws-iot-device-sdk to use version 2.1.3 of MQTT.js.  Note that we haven't done much testing against this version yet – let us know if that works for you, and thanks for using AWS IoT!

```json
{
  "name": "test-package",
  "version": "1.0.0",
  "dependencies": {
    "aws-iot-device-sdk": {
      "version": "1.0.13",
      "from": "aws-iot-device-sdk@>=1.0.12 <2.0.0",
      "resolved": "https://registry.npmjs.org/aws-iot-device-sdk/-/aws-iot-device-sdk-1.0.13.tgz",
      "dependencies": {
        "mqtt": {
          "version": "2.1.3",
          "from": "mqtt@2.1.3",
          "resolved": "https://registry.npmjs.org/mqtt/-/mqtt-2.1.3.tgz"
        }
      }
    }
  }
}
```

**Comment by celalo** [comment]
Very glad to hear that you came up with this. Since then, I had to divert my focus on building other parts of the project and I shall come back to this later on. I will ping here once I get my hands on the module again. Again, thank you very much.

**Comment by ajones-miovision** [comment]
The version of mqtt being used by the iot sdk has a flaw that the iot sdk does not handle gracefully. We  patched the mqtt library last year (see https://github.com/mqttjs/MQTT.js/issues/460) but until iot moves to mqtt >= 2.1 users of the iot device sdk may have to deal with frequent disconnections. To be honest, I'm quite surprised more iot customers have not complained about this issue as it is quite glaring. I suppose it's possible that very few customers use the sdk with devices that have weak connections (we are using cell connections in areas where the signal is weak).

**Comment by bradstewart** [comment]
Any update on when you might support an updated mqtt.js?

**Comment by fengsongAWS** [comment]
Fixed in v2.0.0.

**Comment by barryvdh** [comment]
Currently, version 2.2.1 is required, but currently `mqtt` is at `v2.9.0`.
